core/src/main/java/org/elasticsearch/action/support/replication/ReplicationOperation.java

```diff
@@ -78,11 +78,11 @@
 
     private final List<ReplicationResponse.ShardInfo.Failure> shardReplicaFailures = Collections.synchronizedList(new ArrayList<>());
 
-    ReplicationOperation(Request request, Primary<Request, ReplicaRequest, PrimaryResultT> primary,
-                         ActionListener<PrimaryResultT> listener,
-                         boolean executeOnReplicas, boolean checkWriteConsistency,
-                         Replicas<ReplicaRequest> replicas,
-                         Supplier<ClusterState> clusterStateSupplier, ESLogger logger, String opType) {
+    public ReplicationOperation(Request request, Primary<Request, ReplicaRequest, PrimaryResultT> primary,
```

Comment:
we could, problem is that ReplicationOperation is currently part of the universe of actions and is also used by TransportReplicationAction (which is subclassed by TransportIndexAction, for example)  - so moving it will cause problems with those actions. IMO it's not a big deal to make this public?